ios/browser/api/bookmarks/importer/brave_bookmarks_importer.mm

```diff
@@ -0,0 +1,249 @@
+                                      listener)
+                       );
+      } else {
+        listener(BraveBookmarksImporterStateCompleted, [importer convertToIOSImportedBookmarks:bookmarks]);
+      }
+    } else {
+      listener(BraveBookmarksImporterStateCancelled, nullptr);
+    }
+  };
+  
+  // Run the importer on the sequenced task runner.
+  __weak BraveBookmarksImporter* weakSelf = self;
+  import_thread_->PostTask(FROM_HERE,
+                           base::BindOnce(start_import,
+                                          weakSelf,
+                                          source_file_path,
+                                          top_level_folder_name,
+                                          automaticImport,
+                                          listener)
+                           );
+}
+
+- (void)importFromArray:(NSArray<BraveImportedBookmark *> *)bookmarks
+     topLevelFolderName:(NSString *)folderName
+           withListener:(void(^)(BraveBookmarksImporterState))listener {
+  
+  //In Chromium, this is IDS_BOOKMARK_GROUP (804)
+  base::string16 top_level_folder_name =
+      base::SysNSStringToUTF16(folderName);
+  
+  auto start_import = [](BraveBookmarksImporter* weak_importer,
+                         NSArray<BraveImportedBookmark *> *bookmarks,
+                         const base::string16& top_level_folder_name,
+                         std::function<void(BraveBookmarksImporterState)> listener){
+    //Import cancelled as the importer has been deallocated
+    __strong BraveBookmarksImporter* importer = weak_importer;
+    if (!importer) {
+      listener(BraveBookmarksImporterStateStarted);
+      listener(BraveBookmarksImporterStateCancelled);
+      return;
+    }
+    
+    listener(BraveBookmarksImporterStateStarted);
+    BookmarksImporter::AddBookmarks(top_level_folder_name,
+                                    [importer convertToChromiumImportedBookmarks:bookmarks]);
+    listener(BraveBookmarksImporterStateCompleted);
+  };
+  
+  // Import into the Profile/ChromeBrowserState on the main-thread.
+  __weak BraveBookmarksImporter* weakSelf = self;
+  base::PostTask(FROM_HERE,
+                 {web::WebThread::UI},
+                 base::BindOnce(start_import,
+                                weakSelf,
+                                bookmarks,
+                                top_level_folder_name,
+                                listener)
+                );
+}
+
+// MARK: - Private
+
+- (bool)isImporterCancelled {
+    return self.cancelled;
+}
+
+// Returns true if |url| has a valid scheme that we allow to import. We
+// filter out the URL with a unsupported scheme.
+- (bool)canImportURL:(const GURL&)url {
+    // The URL is not valid.
+    if (!url.is_valid()) {
+      return false;
+    }
+
+    // Filter out the URLs with unsupported schemes.
+    const char* const kInvalidSchemes[] = {"wyciwyg", "place", "about", "chrome"};
```

Comment:
this seems like it should be a whitelist of valid schemes rather than a blacklist of invalid schemes. for instance i assume ones like `tel`, `mailto`, and `wss` are invalid.

Reply:
I believe Brave-iOS as well as the iOS system does support `tel:` and `mailto:`

Reply:
interesting, i would have thought it only supported http, https, javascript, and maybe data.